drivechain-server/bdk/bdk.go

```diff
@@ -293,3 +297,166 @@ type Transaction struct {
 	Sent     btcutil.Amount `json:"sent"`
 	TXID     string         `json:"txid"`
 }
+
+// NewWallet either creates a new wallet or loads an existing one. If a
+// passphrase is provided, the mnemonic is encrypted before saving if the
+// wallet is created. If the wallet already exists, the passphrase is used to
+// decrypt the existing mnemonic.
+func NewWallet(datadir, network, electrum, passphrase string) (*Wallet, error) {
```

Comment:
This isn't idempotent and _could_ be refactored, imo not worth it at the time but open for consideration.